src/internal/_assocPath.js

```diff
@@ -0,0 +1,11 @@
+var _assoc = require('./_assoc');
+var _slice = require('./_slice');
+
+
+module.exports = function _assocPath(path, val, obj) {
```

Comment:
> On the other hand one could argue that since Ramda promotes programming without mutation, cloning is unnecessary.

That's the argument I would make.  I don't see a reason to treat the top-level object any different from the sub-object properties, which are references.